### ISSUE TITLE
KeyError: 'PYTEST_CURRENT_TEST' while sending POST/PUT requests to RP

### ISSUE THREAD

**Issue by ilia-tsyplenkov** [reportportal/client-Python#94]
https://github.com/reportportal/client-Python/issues/39 fixed only one problem case.
To finally fix race condition issue need to handle all possible cases which are all HTTP requests via `request` package since there is no guarantee that environ dict wouldn't be modified by other service
There are next events which have same problem:
- `reportportal_client/service.py::ReportPortalService::start_launch`
- `reportportal_client/service.py::ReportPortalService::finish_launch`
- `reportportal_client/service.py::ReportPortalService::start_test_item`
- `reportportal_client/service.py::ReportPortalService::update_test_item`
- `reportportal_client/service.py::ReportPortalService::finish_test_item`
- `reportportal_client/service.py::ReportPortalService::log`
- `reportportal_client/service.py::ReportPortalService::get_item_id_by_uuid`
- `reportportal_client/service.py::ReportPortalService::get_project_settings`

I regulary have fails with:
- `reportportal_client/service.py::ReportPortalService::start_test_item`
- `reportportal_client/service.py::ReportPortalService::finish_test_item`

```File /usr/local/lib/python3.6/dist-packages/reportportal_client/service_async.py, line 234, in process_item.    getattr(self.rp_client, method)(**kwargs).
File /usr/local/lib/python3.6/dist-packages/reportportal_client/service.py, line 229, in finish_test_item.    r = self.session.put(url=url, json=data, verify=self.verify_ssl).
File /usr/local/lib/python3.6/dist-packages/requests/sessions.py, line 590, in put.    return self.request(PUT, url, data=data, **kwargs).
File /usr/local/lib/python3.6/dist-packages/requests/sessions.py, line 521, in request.    prep.url, proxies, stream, verify, cert.
File /usr/local/lib/python3.6/dist-packages/requests/sessions.py, line 697, in merge_environment_settings.    env_proxies = get_environ_proxies(url, no_proxy=no_proxy).
File /usr/local/lib/python3.6/dist-packages/requests/utils.py, line 766, in get_environ_proxies.    if should_bypass_proxies(url, no_proxy=no_proxy):.
File /usr/local/lib/python3.6/dist-packages/requests/utils.py, line 750, in should_bypass_proxies.    bypass = proxy_bypass(parsed.hostname).
File /usr/lib/python3.6/urllib/request.py, line 2497, in proxy_bypass_environment.    proxies = getproxies_environment().
File /usr/lib/python3.6/urllib/request.py, line 2470, in getproxies_environment.    for name, value in os.environ.items():.
File /usr/lib/python3.6/_collections_abc.py, line 744, in __iter__.    yield (key, self._mapping[key]).
File /usr/lib/python3.6/os.py, line 669, in __getitem__.    raise KeyError(key) from None.KeyError: PYTEST_CURRENT_TEST
```

I'm going to prepare PR if you don't mind.
Could you please grant me write permissions?

Please note that it is very important for my team to provide fix for this issue in https://github.com/reportportal/client-Python/releases/tag/3.2.3 too since can't to work with the latest report portal agent/ client right now

**Comment by ilia-tsyplenkov** [comment]
@iivanou I don't have write permission in this repo to push. Could you grant it to me?

**Comment by iivanou** [comment]
You should not push to this repo. Create your fork off the master and push the changes there.

**Comment by iivanou** [comment]
Fixed in #96.
